scripts/python/parthenon_metrics_app.py

```diff
@@ -0,0 +1,791 @@
+
+    This class is responsible for reading performance data from json files.
+    """
+
+    def _containsCommit(self, json_objs, value):
+        """Will determine if a commit is found in the performance json files."""
+        if not isinstance(json_objs, list):
+            json_objs = [json_objs]
+
+        for json_obj in json_objs:
+            if json_obj.get('commit sha') == value:
+                return True
+
+        return False
+
+    def _add_to_json_obj(self, new_data):
+        """json data should be of the following form:
+
+            Where there are blocks of data in a list with the content below
+
+                    'commit sha': commit_sha,
+                    'branch': current_branch,
+                    'date': now.strftime("%Y-%m-%d %H:%M:%S"),
+                    'data':[{
+                      'test': dir,
+                      'mesh_blocks': mesh_blocks,
+                      'zone_cycles': zone_cycles}]
+        """
+        # Cycle the outer list first
+        if isinstance(self._data, list):
+            for json_obj in self._data:
+                if json_obj.get('commit sha') == new_data.get('commit sha'):
+                    for data_grp in json_obj.get('data'):
+                        for data_grp2 in new_data.get('data'):
+                            if data_grp.get('test') == new_data.get('test'):
+                                # Overwrite the existing content with the new
+                                # content
+                                data_grp['mesh_blocks'] = copy.deepcopy(
+                                    data_grp2.get('mesh_blocks'))
+                                data_grp['zone_cycles'] = copy.deepcopy(
+                                    data_grp2.get('zone_cycles'))
+                    return
+                else:
+                    # Then the test was not found so we are going to append to
+                    # it
+                    json_obj['data'].append(new_data['data'])
+        else:
+            if isinstance(new_data, list):
+                if len(new_data) == 1:
+                    new_data = new_data[0]
+
+            if self._data.get('commit sha') == new_data.get('commit sha'):
+                for data_grp in self._data.get('data'):
+                    for data_grp2 in new_data.get('data'):
+                        if data_grp.get('test') == data_grp2.get('test'):
+                            # Overwrite the existing content with the new
+                            # content
+                            data_grp['mesh_blocks'] = copy.deepcopy(
+                                data_grp2.get('mesh_blocks'))
+                            data_grp['zone_cycles'] = copy.deepcopy(
+                                data_grp2.get('zone_cycles'))
+                            return
+            else:
+                # Then the test was not found so we are going to append to it
+                self._data['data'].append(new_data['data'])
```

Comment:
Would this work? Please double check where the `return` should go.

```suggestion
        if isinstance(new_data, list):
            if len(new_data) == 1:
                new_data = new_data[0]
            else:
                raise ValueError("Expected exactly 1 new data")
                
        dat_list = self._data if isinstance(self._data, list) else [self._data]
        for json_obj in dat_list:
            if json_obj.get('commit sha') == new_data.get('commit sha'):
                for data_grp in json_obj.get('data'):
                    for data_grp2 in new_data.get('data'):
                        if data_grp.get('test') == new_data.get('test'):
                            # Overwrite the existing content with the new content
                            data_grp['mesh_blocks'] = copy.deepcopy(
                                data_grp2.get('mesh_blocks'))
                            data_grp['zone_cycles'] = copy.deepcopy(
                                data_grp2.get('zone_cycles'))
                return
            else:
                # Then the test was not found so we are going to append to it
                json_obj['data'].append(new_data['data'])
```

Reply:
Done